### PR TITLE
You can no longer enter jaunt in NOJAUNT areas

### LINKED ISSUE #56603: Wizards can jaunt out of the wizard ship and onto the CentCom z-level.
## Reproduction:
1. Become a space wizard
2. Jaunt
3. Unjaunt
4. Teleport to somewhere on the CentCom z-level.

https://user-images.githubusercontent.com/53582584/106683945-56b43c00-657a-11eb-8ce7-6bbe54eb0a7f.mp4

@tralezab

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

You could use the game trying to send you away from a NOJAUNT area to get on centcom!

## Why It's Good For The Game

#fixes #56603

The intended process is to send people away IF they start a jaunt and then end it in a NOJAUNT, aka going out of your way to try and cheat into nojaunt

## Changelog
:cl:
fix: No more centcom jaunting!
/:cl:

### CODE REVIEW

File: code/modules/spells/spell_types/bloodcrawl.dm [hunk 1]
```diff
@@ -14,6 +14,15 @@
 	action_background_icon_state = "bg_demon"
 	var/phased = FALSE
 
+/obj/effect/proc_holder/spell/bloodcrawl/cast_check(skipcharge = 0,mob/user = usr)
+	. = ..()
+	if(!.)
+		return FALSE
+	var/area/noteleport_check = get_area(user)
+	if(noteleport_check && noteleport_check.area_flags & NOTELEPORT)
+		to_chat(user, "<span class='danger'>Some dull, universal force is inbetween you and your other existence, preventing you from blood crawling.</span>")
```
Comment (ATH1909) [inline code/modules/spells/spell_types/bloodcrawl.dm:23]: "inbetween"

File: code/modules/spells/spell_types/bloodcrawl.dm [hunk 2]
```diff
@@ -14,6 +14,15 @@
...
 	var/phased = FALSE
 
+/obj/effect/proc_holder/spell/bloodcrawl/cast_check(skipcharge = 0,mob/user = usr)
+	. = ..()
+	if(!.)
+		return FALSE
+	var/area/noteleport_check = get_area(user)
+	if(noteleport_check && noteleport_check.area_flags & NOTELEPORT)
+		to_chat(user, "<span class='danger'>Some dull, universal force is between you and your other existence, preventing you from blood crawling.</span>")
+		return FALSE
```
Comment (MrDoomBringer) [inline code/modules/spells/spell_types/bloodcrawl.dm:24]: I feel like you could compress these few lines of repeated code into a more generic helper function

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

File: code/modules/spells/spell_types/bloodcrawl.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/spells/spell_types/bloodcrawl.dm:24]: Honestly best solution would be combining all JAUNTS into one.

Comment (MrDoomBringer) [reply code/modules/spells/spell_types/bloodcrawl.dm:24]: yes but until then ;)

**Comment by Jared-Fogle** [comment]
merge from upstream plz

**Pull request merged**